openml/_api_calls.py

```diff
@@ -174,11 +186,14 @@ def __read_url(url, request_method, data=None):
 def _send_request(
     request_method, url, data, files=None,
 ):
-    n_retries = config.connection_n_retries
+    n_retries = max(1, min(config.connection_n_retries, config.max_retries))
+
+    retry_counter = 0
     response = None
     with requests.Session() as session:
         # Start at one to have a non-zero multiplier for the sleep
-        for i in range(1, n_retries + 1):
+        while retry_counter < n_retries:
```

Comment:
This looks like a `for retry_counter in range(n_retries)` loop with extra steps to me.

Reply:
It was changed to a `while` since an intermediate change made it easier to work with rather than a `for`. However, with the latest changes in this loop, it can be reverted back to a `for`.